Source/ToolCore/Build/ResourcePackager.cpp

```diff
@@ -201,11 +201,6 @@ void ResourcePackager::GeneratePackage(const String& destFilePath)
             return;
         }
 
-        if (!file.GetSize())
-        {
-            return;
-        }
-
```

Comment:
Yeah, TBH Unity's native handling of different builds is pretty weak. It's _ok_ at handling things on different platforms, but realistically today that's just not enough. As an example on Android, you need to be able to target different chipsets with different prebuilt texture sets and now even on iOS there's native support for addon resources and if I'm not mistaken they can be refined based on device capabilities. You want to be able to have builds tailored for each of those. First step towards that would be a build that takes a list of resources to include, second would be a file format for that list. 

On the editor side, lots of possible options for making it easy to control how that gets generated, I suppose looking at what others do as a start would be great and then having a serious think about which way you want to go.